enigma/src/main/java/org/quiltmc/enigma/api/EnigmaProject.java

```diff
@@ -167,6 +169,68 @@ private Collection<Entry<?>> dropMappings(EntryTree<EntryMapping> mappings, Prog
 		return dropper.getDroppedMappings().keySet();
 	}
 
+	public Collection<Entry<?>> dropProposedMappings(ProgressListener progress) {
+		DeltaTrackingTree<EntryMapping> mappings = this.remapper.getMappings();
+		EntryTree<EntryMapping> proposedMappings = this.remapper.getProposedMappings();
+		progress.init((int) proposedMappings.getAllEntries().count(), "Proposed Mappings to check");
+
+		List<Entry<?>> dropped = new ArrayList<>();
+
+		int steps = 0;
+		for (EntryTreeNode<EntryMapping> proposedMapping : proposedMappings) {
+			progress.step(steps++, proposedMapping.getEntry().getFullName());
+
+			if (proposedMapping.getValue() == null) {
+				continue;
+			}
+
+			if (!mappings.contains(proposedMapping.getEntry())) {
+				continue;
+			}
+
+			EntryMapping entryMapping = mappings.get(proposedMapping.getEntry());
+
+			if (entryMapping.tokenType().isProposed()) {
+				continue;
+			}
+
+			if (entryMapping.targetName() == null) {
+				continue;
+			}
+
+//			if (entryMapping.javadoc() != null) {
+//				continue;
+//			}
+
+//			ParentedEntry<?> parent = ((ParentedEntry<?>) proposedMapping.getEntry());
+//			while (parent != null && parent.getParent() != null) {
+//				parent = (ParentedEntry<?>) parent.getParent();
+//			}
+//
+//			if (!parent.getFullName().equals("net/minecraft/unmapped/C_ygpuayyc")) {
+//				continue;
+//			}
+
+			if (proposedMapping.getValue().sourcePluginId().equals("quiltmc:name_proposal/delegate_params")) {
```

Comment:
this should preserve fallback-proposed names as well
also why do delegate params need an exception?

Reply:
I believe this was a work around until we figure out why we are getting the massive spooky diffs,

Reply:
ah, ok
it could be a useful to have an arg for proposer ids to exclude

Reply:
potentially, but i also feel like that defeats a bit of the purpose of this feature, which is to make sure that no more proposed names are left in the mappings. If the proposer is fragile enough to break with this, its probably not written very well.

Reply:
true
I wouldn't expect it to be used often
maybe it would be better not to have an arg; if a proposer is having issues it can just be set to fallback as a workaround